pkg/api/dirs.go

```diff
@@ -117,24 +117,20 @@ func storeDir(ctx context.Context, reader io.ReadCloser, s storage.Storer, logge
 		// create manifest entry for uploaded file
 		headers := http.Header{}
 		headers.Set("Content-Type", contentType)
-		fileEntry := &jsonmanifest.JSONEntry{
-			Reference: fileReference,
-			Name:      fileName,
-			Headers:   headers,
-		}
+		fileEntry := jsonmanifest.NewEntry(fileReference, fileName, headers)
 
 		// add entry to dir manifest
 		dirManifest.Add(filePath, fileEntry)
 	}
 
 	// check if files were uploaded through the manifest
-	if len(dirManifest.Entries) == 0 {
-		return swarm.ZeroAddress, fmt.Errorf("no files in tar")
+	if dirManifest.Size() == 0 {
```

Comment:
I think `Length` is more legible here as `Size` might infer actual size in bytes

Reply:
agreed, done